Axinom/mosaic-media-template
# [AB#41880] localizable properties are embedded into the root entity catalog api with views

# Pull Request

## Prerequisites

      releases)

## Description

Instead of having a `localization` sub-object for each localizable entity, localizable properties are now embedded into it.
The filtering by locale is happening via the headers not to break "[the contract that a given identified entity is always the same entity](https://discord.com/channels/489127045289476126/498852330754801666/1166690921547571240)". 

This implementation makes it more convenient for the FE apps that would be using the Catalog Service API, since there will be no need for them to know which property is localizable, and which is not. This is the most consistent and performant approach. [The alternative](https://github.com/Axinom/mosaic-media-template/pull/187) is a bit easier to define, but has an N+1 issue.
With Postgraphile V5, this approach will most probably change.

The GraphQL schema changes are just re-ordering. Schema is the same as currently defined in `dev` branch.

## Testing notes
- Try testing the catalog API and make sure that content is returned based on the passed header parameter, e.g.
```
{
"mosaic-locale": "de-DE"
}
```
- The general behavior should remain the same (e.g. if locale exists - return the localized value. If it does not exist or not specified - base value is returned). Localizable properties include title, description, synopsis. Channel has no synopsis, Season has no title, Genres have only title.
- Make sure that relations are properly returned (e.g. episodes for season, related items for collection, images, licenses, videos, genres for most items, etc...)
